File: problem/power/6/main.py

# main.py: main file for "power" solutions
#
# Copyright (C) 2020  Rudy Matela
# All rights reserved.
#
#
# This file is part of the CSCX Online Judge.
#
# This is to be appended to the submitted file.

# standard input is already processed
# as the submitted programmed is run when imported

with open("in.txt") as filein:
  for line in filein:
    b, e = line.split()
    b = int(b)
    e = int(e)
    print(power(b, e))

File: problem/power/6/main.hs
-- main.hs: main file for "power" solutions
--
-- Copyright (C) 2020  Rudy Matela
-- All rights reserved.
--
--
-- This file is part of the CSCX Online Judge.
--
-- This is to be linked to the submitted file.
-- It processes values from standard input then from the "in.txt" file.
module AltMain (main) where

import Main (power)

io1 :: String -> String
io1  =  show . (\[b,e] -> power b e) . map read . words

io :: String -> String
io = unlines . map io1 . lines

main :: IO ()
main = do
  -- processes standard input
  interact io

  -- processes in.txt
  more <- readFile "in.txt"
  putStr $ io more
